packages/sitecore-jss/src/utils/utils.test.ts

```diff
@@ -181,4 +182,61 @@ describe('utils', () => {
       expect(isTimeoutError({ name: 'AbortError' })).to.be.true;
     });
   });
+
+  describe('enforceCors', () => {
+    const mockOrigin = 'https://maybeallowed.com';
+    const mockRequest = (origin?: string) => {
+      return {
+        headers: {
+          origin: origin || mockOrigin,
+        },
+      } as IncomingMessage;
+    };
+
+    const mockResponse = () => {
+      const res = {} as OutgoingMessage;
+      res.setHeader = spy(() => {
+        return res;
+      });
+
+      return res;
+    };
+
+    it('should return true if origin is found in allowedOrigins from API_ALLOWED_ORIGINS env variable', () => {
+      const req = mockRequest();
+      const res = mockResponse();
+      process.env.API_ALLOWED_ORIGINS = mockOrigin;
+      expect(enforceCors(req, res)).to.be.equal(true);
+      delete process.env.API_ALLOWED_ORIGINS;
+    });
+
+    it('should return true if origin is found in allowedOrigins passed as argument', () => {
+      const req = mockRequest('http://allowed.com');
+      const res = mockResponse();
+
+      expect(enforceCors(req, res, ['http://allowed.com'])).to.be.equal(true);
+    });
+
+    it('should return false if origin matches neither allowedOrigins from API_ALLOWED_ORIGINS env variable nor argument', () => {
+      const req = mockRequest('https://notallowed.com');
+      const res = mockResponse();
+      process.env.API_ALLOWED_ORIGINS = 'https://strictallowed.com, https://alsoallowed.com';
+      expect(enforceCors(req, res, ['https://paramallowed.com'])).to.be.equal(false);
+      delete process.env.API_ALLOWED_ORIGINS;
+    });
+
+    it('should return true when origin matches a wildcard value from allowedOrigins', () => {
```

Comment:
can you please check if this will pas with origin 'https://pagesdev.cloud' and allowed origins 'http://pages*.cloud' ?

Reply:
I can, but it would essentially be the same test, wouldn't it?
I tested 'pages*' origins matching manually - it works.